### ISSUE TITLE
iOS 18 update - support for "select contacts" flow

### ISSUE THREAD

**Issue by emily-ginsburg** [capacitor-community/contacts#125]
**Is your feature request related to a problem? Please describe.**
iOS 18 added the option to select which contacts you share with an app (instead of just "allow all" or "deny all"). When a user selects the option to select individual contacts for sharing, this plugin doesn't indicate that the permission is different (result of `requestPermissions` is `{contacts: 'granted'}`), and there's no way to prompt the user to select more contacts to share (`requestPermissions` and `getContacts` behavior not changed, as if all my contacts are already shared).

**Describe the solution you'd like**
I would expect an additional response option in `PermissionState` that indicates only some contacts were shared (e.g. `Geolocation.requestPermissions` from `@capacitor/geolocation` returns `PermissionStatus` that includes two states, precise location and coarse location). And then I would expect there to be a function that allows the app to prompt the user to share more contacts (e.g. `Camera.pickImages` from `@capacitor/camera` when individual photos are shared)

**Describe alternatives you've considered**
Using an alternate plugin I guess :')

**Additional context**
![IMG_0901](https://github.com/user-attachments/assets/8ddedbc3-3fa0-4f22-a64d-ace164ba6888)

**Comment by daniloargentiero** [comment]
I encountered the same issue. Please prioritize this bug as it makes the plugin incompatible with iOS 18.

**Comment by daniloargentiero** [comment]
I've created a pull request: https://github.com/capacitor-community/contacts/pull/137

**Comment by stevecarlton** [comment]
@tafelnl can we please have @daniloargentiero's pull request #137 added? The plugin is essentially broken for many iOS 18 users at this point. I have tested the pull request and have found no issues.

**Comment by stephan-fischer** [comment]
Meanwhile i forked and merged it:
https://github.com/stephan-fischer/capacitor-contacts

**Comment by gregzo** [comment]
Bumping this, without pull request #137 the plugin is not usable on iOS 18+.

**Comment by timmbach** [comment]
Did you discover a solution to this? @emily-ginsburg

**Comment by emily-ginsburg** [comment]
> Did you discover a solution to this? [@emily-ginsburg](https://github.com/emily-ginsburg)

no, would be great to get @daniloargentiero 's pr merged if it is working. would really like to use the main branch instead of a forked one! https://github.com/capacitor-community/contacts/pull/137

**Comment by timmbach** [comment]
I guess i'll just fork his branch too then.

**Comment by timmbach** [comment]
Even a fresh install (on both ios and android) shows denied by default, it doesnt even let me see the requestPermission popup.

**Comment by tafelnl** [comment]
The code from `requestPermission` is not doing anything custom. This is the code:

```swift
    @objc override public func requestPermissions(_ call: CAPPluginCall) {
        CNContactStore().requestAccess(for: .contacts) { [weak self] _, _  in
```

so, again, it seems like a configuration issue on your end. If your issue persists, please create a new issue and provide a [Minimal, Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example) using a minimal Capacitor app in a public GitHub repository
